rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -74,7 +74,8 @@ void TimeSource::attachNode(
 
   auto cb = std::bind(&TimeSource::clock_cb, this, std::placeholders::_1);
 
-  clock_subscription_ = rclcpp::create_subscription<MessageT, decltype(cb), Alloc, SubscriptionT>(
+  clock_subscription_ = rclcpp::create_subscription<
+    MessageT, decltype(cb), Alloc, MessageT, SubscriptionT>(
```

Comment:
This is bad style in my opinion because the template arguments line up with the function arguments. Instead do:

```c++
  clock_subscription_ = rclcpp::create_subscription<
    MessageT, decltype(cb), Alloc, MessageT, SubscriptionT
  >(
    node_topics_.get(),
    topic_name,
    std::move(cb),
    ...
  );
```

Or:

```c++
  clock_subscription_ =
    rclcpp::create_subscription<MessageT, decltype(cb), Alloc, MessageT, SubscriptionT>(
      node_topics_.get(),
      topic_name,
      std::move(cb),
      ...
    );
```

Reply:
Both suggest versions don't work, but I'll go for this one then to separate the template arguments from the function arguments:
```
  clock_subscription_ = rclcpp::create_subscription<
    MessageT, decltype(cb), Alloc, MessageT, SubscriptionT
    >(
    node_topics_.get(),
    topic_name,
    std::move(cb),
    rmw_qos_profile_default,
    group,
    false,
    false,
    msg_mem_strat,
    allocator);

```